lib/hl7/message.ex

```diff
@@ -529,6 +526,30 @@ defmodule HL7.Message do
     end
   end
 
+  defp maybe_transcode(binary) do
+    case String.valid?(binary) do
+      true ->
+        binary
+
+      false ->
+        transcode(binary)
+    end
+  end
+
+  defp transcode(binary) do
+    transcode(binary, [])
+  end
+
+  defp transcode(<<>>, acc), do: acc |> Enum.reverse() |> to_string()
```

Comment:
Between this version and basically how String.valid/1 is implemented, it may be best just to go with the String.valid/1 approach since it's simpler. No strong feelings.

Reply:
Please benchmark but I believe the Erlang VM can optimize the binary one really well.

Reply:
Your simpler implementation is faster about 14% percent faster. ❤️